src/sections/provider/components/ProviderItem/ProviderItem.tsx

```diff
@@ -150,38 +150,45 @@ const ProviderSelectItemExternal = ({
 
   useEffect(() => {
     const rpc = url
-    const provider = new WsProvider(rpc)
-
-    let cancel: () => void
-
-    async function load() {
-      const apiPool = SubstrateApis.getInstance()
-      const api = await apiPool.api(provider.endpoint)
-
-      async function onNewBlock() {
-        const [parachain, timestamp] = await Promise.all([
-          api.derive.chain.bestNumber(),
-          api.query.timestamp.now(),
-        ])
-
-        setBestNumberState({
-          parachainBlockNumber: parachain,
-          timestamp: timestamp,
-        })
-      }
-
-      api.on("connected", onNewBlock)
-      api.rpc.chain
-        .subscribeNewHeads(onNewBlock)
-        .then((newCancel) => (cancel = newCancel))
-    }
-
-    load()
 
-    return () => {
-      cancel?.()
-      provider.disconnect()
+    const getProvider = async () => {
+      try {
+        const provider = await connectWsProvider(rpc)
+
+        let cancel: () => void
+
+        async function load() {
+          const apiPool = SubstrateApis.getInstance()
+
+          const api = await apiPool.api(provider.endpoint)
+          async function onNewBlock() {
+            const [parachain, timestamp] = await Promise.all([
+              api.derive.chain.bestNumber(),
+              api.query.timestamp.now(),
+            ])
+
+            setBestNumberState({
+              parachainBlockNumber: parachain,
+              timestamp: timestamp,
+            })
+          }
+
+          api.on("connected", onNewBlock)
+          api.rpc.chain
+            .subscribeNewHeads(onNewBlock)
+            .then((newCancel) => (cancel = newCancel))
+        }
+
+        load()
+
+        return () => {
+          cancel?.()
+          provider.disconnect()
+        }
+      } catch (error) {}
     }
+
+    getProvider()
```

Comment:
```suggestion
    return getProvider()
```

since getProvider is function returning a cleanup function, you should probably return it in useEffect, so the cleanup function is called on unmount